build: use macOS 14 SDK (Xcode 15.0)

fanquake opened the pull request:
This fixes: https://github.com/bitcoin/bitcoin/pull/28349#issuecomment-1748515277 (cross-compiling with C++20 for macOS). See https://developer.apple.com/xcode/cpp/#c++20 for C++20 support in Apples libc++, some features landed with Xcode 14.3, although many more landed with Xcode 15.0.

fanquake committed:
build: use macOS 14 SDK (Xcode 15.0)

hebasto commented:
Concept ACK.

hebasto commented:
> Does this affect which versions of macos the app can be run on?

It depends on the `-mmacosx-version-min` option?

Sjors commented on contrib/macdeploy/README.md:
```diff
@@ -14,51 +14,45 @@ When complete, it will have produced `Bitcoin-Core.zip`.
 
-The `sha256sum` of the generated TAR.GZ archive should be `df75d30ecafc429e905134333aeae56ac65fac67cb4182622398fd717df77619`.
+The generated archive should be: `Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz`.
+The `sha256sum` should be `50073e35975a5336f7db3abb275c00b372e07ff2372d7e78508c722abd746b9f`.
```
On (Intel) macOS Ventura 13.6 and on Ubuntu 23.10 I get a different checksum:

```
c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```

[sums.txt](https://github.com/bitcoin/bitcoin/files/12882430/sums.txt)

(note to reviewers: be very careful to delete the extracted `Xcode.app` folder before extracting the zip, or you _may_ get confused)

fanquake replied:
Can you retry with the new script.

Sjors replied:
Will do soon(tm)

Sjors commented:
Still getting a different hash when using the gen-sdk script in f638625ad416f9749633925bec827e29b5ea3f79:

```
70cccc19df2fc1746bfddf939bf7370eb0b3d8f5f7198896358fa1ebde37e9e7  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```

Both on macOS and Ubuntu.

josibake commented:
I'm getting the same as @Sjors :

```
70cccc19df2fc1746bfddf939bf7370eb0b3d8f5f7198896358fa1ebde37e9e7  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```

fanquake commented:
Updated the hash. Also going to test the 15.1 beta, to see if that fixes the Qt build issues.

hebasto commented on contrib/macdeploy/gen-sdk:
```diff
@@ -99,16 +96,15 @@ def run():
             # recursion already adds entries in sorted order
-            tarfp.add(".", recursive=True, filter=change_tarinfo_base)
+            tarfp.add("./usr/include", recursive=True, filter=change_tarinfo_base)
+            tarfp.add("./usr/lib", recursive=True, filter=change_tarinfo_base)
```
This change causes `fatal error: 'CoreFoundation/CoreFoundation.h' file not found` when building the `qt` package in depends.

fanquake replied:
Looks like we have to retain `System/Library/` for Qt. Will return in the next push.

fanquake commented:
> Also going to test the 15.1 beta, to see if that fixes the Qt build issues.

No difference with Xcode 15.1-beta2. Qt still fails to build the same way as described in the PR description.

hebasto commented:
> Qt still fails to build the same way as described in the PR description.

The following patch fixes the Qt build for me:
```diff
--- a/depends/hosts/darwin.mk
+++ b/depends/hosts/darwin.mk
@@ -1,4 +1,5 @@
 OSX_MIN_VERSION=11.0
+_OSX_MIN_VERSION=110000
 OSX_SDK_VERSION=14.0
 XCODE_VERSION=15.0.1
 XCODE_BUILD_ID=15A507
@@ -75,7 +76,7 @@ $(foreach TOOL,$(cctools_TOOLS),$(eval darwin_$(TOOL) = $$(build_prefix)/bin/$$(
 darwin_CC=env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH \
               -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH \
               -u LIBRARY_PATH \
-            $(clang_prog) --target=$(host) -mmacosx-version-min=$(OSX_MIN_VERSION) \
+            $(clang_prog) --target=$(host) -mmacosx-version-min=$(OSX_MIN_VERSION) -D__MAC_OS_X_VERSION_MIN_REQUIRED=$(_OSX_MIN_VERSION) \
               -B$(build_prefix)/bin -mlinker-version=$(LD64_VERSION) \
               -isysroot$(OSX_SDK) -nostdlibinc \
               -iwithsysroot/usr/include -iframeworkwithsysroot/System/Library/Frameworks
@@ -83,7 +84,7 @@ darwin_CC=env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH \
 darwin_CXX=env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH \
                -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH \
                -u LIBRARY_PATH \
-             $(clangxx_prog) --target=$(host) -mmacosx-version-min=$(OSX_MIN_VERSION) \
+             $(clangxx_prog) --target=$(host) -mmacosx-version-min=$(OSX_MIN_VERSION) -D__MAC_OS_X_VERSION_MIN_REQUIRED=$(_OSX_MIN_VERSION) \
                -B$(build_prefix)/bin -mlinker-version=$(LD64_VERSION) \
                -isysroot$(OSX_SDK) -nostdlibinc \
                -iwithsysroot/usr/include/c++/v1 \
```

However, the underlying issue still needs to be investigated further.

fanquake commented:
> The following patch fixes the Qt build for me:

We wont be taking that patch, but if the problem is that qt fails to respect `-mmacosx-version-min` for some reason, I would assume that should make it fairly obvious where the problem is.

hebasto commented:
> > The following patch fixes the Qt build for me:
> 
> We wont be taking that patch, but if the problem is that qt fails to respect `-mmacosx-version-min` for some reason, I would assume that should make it fairly obvious where the problem is.

Apparently, direct tests show that `env -u C_INCLUDE_PATH -u CPLUS_INCLUDE_PATH -u OBJC_INCLUDE_PATH -u OBJCPLUS_INCLUDE_PATH -u CPATH -u LIBRARY_PATH /home/hebasto/git/bitcoin/depends/x86_64-apple-darwin/native/bin/clang++ --target=x86_64-apple-darwin -mmacosx-version-min=11.0 -B/home/hebasto/git/bitcoin/depends/x86_64-apple-darwin/native/bin -mlinker-version=711 -isysroot/home/hebasto/SDKs/Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers -nostdlibinc -iwithsysroot/usr/include/c++/v1 -iwithsysroot/usr/include -iframeworkwithsysroot/System/Library/Frameworks -pipe -std=c++17` does not define the `__MAC_OS_X_VERSION_MIN_REQUIRED` macro.

hebasto commented on depends/hosts/darwin.mk:
```diff
@@ -1,7 +1,7 @@
-OSX_SDK_VERSION=11.0
-XCODE_VERSION=12.2
-XCODE_BUILD_ID=12B45b
+OSX_SDK_VERSION=14.0
+XCODE_VERSION=15.0
+XCODE_BUILD_ID=15A240d
```
According to https://developer.apple.com/support/xcode/, additionally, `OSX_MIN_VERSION=13.5` should be adjusted as well.

hebasto commented:
The PR description looks outdated after the recent push :)

fanquake commented:
Going to wait and see if it actually works first.

hebasto commented:
Is new macOS SDK available at https://bitcoincore.org/depends-sources/sdks/?

fanquake commented:
Nothing has been uploaded yet. I don't think there would be any real difference, but 15.0.1 has been released, and 15.1 is currently in beta 2, so we may pick a slightly different SDK.

hebasto commented:
> Nothing has been uploaded yet. I don't think there would be any real difference, but 15.0.1 has been released, and 15.1 is currently in beta 2, so we may pick a slightly different SDK.

I mean, that CI fails due to not being able to load the SDK...

fanquake commented:
Yea, anyone that wants to test should create the new SDK, check the new shasum, and then do a Guix build. If that works, the CI will.

fanquake commented:
Looks like we'll have to add a `-Wl,-platform_version` into ldflags here.

fanquake commented:
> The PR description looks outdated after the recent push :)

Updated the description, and rebased on #28880 instead, now that we can do that.

fanquake commented on contrib/devtools/symbol-check.py:
```diff
@@ -236,7 +236,7 @@ def check_MACHO_min_os(binary) -> bool:
 
 def check_MACHO_sdk(binary) -> bool:
-    if binary.build_version.sdk == [11, 0, 0]:
+    if binary.build_version.sdk == [14, 0, 0]:
```
The SDK version in a Guix built bitcoind is still being set to 11.0:
```bash
Load command 11
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform 1
    minos 11.0
      sdk 11.0
   ntools 1
     tool 3
  version 711.0
```

fanquake replied:
> Looks like we'll have to add a -Wl,-platform_version into ldflags here.

Should be resolved now.

fanquake committed:
depends: add -platform_version to macOS build flags

```bash
-platform_version platform min_version sdk_version
    This is set to indicate the platform, oldest supported version
    of that platform that output is to be used on, and the SDK that
    the output was built against.
```

hebasto commented:
My SDK hash:
```
$ sha256sum Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz 
6afb7c24461729167c5f5976ce56e774657a97b052f44f1618e9717da67fddb7  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```
and Guix builds:
```
x86_64
fa1db57217fe8646c529f312ba5af6fee9f3085768d2033359b2f1ec9f1c4624  guix-build-cf8353a9cf9c/output/arm64-apple-darwin/SHA256SUMS.part
e64ee8bf9901e5a37cbb67cb5ed23daaa7d0086b256b64870b26a94d5ac3ee4e  guix-build-cf8353a9cf9c/output/arm64-apple-darwin/bitcoin-cf8353a9cf9c-arm64-apple-darwin-unsigned.tar.gz
398fe7d07c783bd9a9f5649946e4f98379fe8aebaa4bd0446b019d70391148af  guix-build-cf8353a9cf9c/output/arm64-apple-darwin/bitcoin-cf8353a9cf9c-arm64-apple-darwin-unsigned.zip
f12ae881d2704b79f038275f53f8e035a33d56ab4128574625812df01b6dceea  guix-build-cf8353a9cf9c/output/arm64-apple-darwin/bitcoin-cf8353a9cf9c-arm64-apple-darwin.tar.gz
81b27791a4d38f503bc7971de905b026e537daaa08b69e6cda890b7e117644f2  guix-build-cf8353a9cf9c/output/dist-archive/bitcoin-cf8353a9cf9c.tar.gz
fd436ffe66254b4d321398b6250d8b8d43b8b9b8c85527363351e92b2f3ecdba  guix-build-cf8353a9cf9c/output/x86_64-apple-darwin/SHA256SUMS.part
194ffdad2f243a7cc8c4498518f48962cbf5fe372eb957f6dfc78620dad6c795  guix-build-cf8353a9cf9c/output/x86_64-apple-darwin/bitcoin-cf8353a9cf9c-x86_64-apple-darwin-unsigned.tar.gz
d3c807b22291e7b168d3d3e8c89dbf1b8c19e638b4a2f2b4ee24993d11081a5a  guix-build-cf8353a9cf9c/output/x86_64-apple-darwin/bitcoin-cf8353a9cf9c-x86_64-apple-darwin-unsigned.zip
209aaeac2b3b6ca0550f6bb53c9a489ee4e0b6a9d2c71e76e9fc3df474eb5337  guix-build-cf8353a9cf9c/output/x86_64-apple-darwin/bitcoin-cf8353a9cf9c-x86_64-apple-darwin.tar.gz
```

There is a mismatch for `arm64-apple-darwin`.

fanquake commented:
Rebased for #28851 and dropped the LLVM changes.

fanquake commented:
`4daaed2ef2253c9661779fa40bfff50655dc7ec45801aba5a39653e7bcdde48e  Xcode_15.xip`.
`c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz`.

Sjors commented on contrib/macdeploy/README.md:
```diff
@@ -14,51 +14,45 @@ When complete, it will have produced `Bitcoin-Core.zip`.
 
-The `sha256sum` of the generated TAR.GZ archive should be `df75d30ecafc429e905134333aeae56ac65fac67cb4182622398fd717df77619`.
+The generated archive should be: `Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz`.
+The `sha256sum` should be `c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d`.
```
This matches for me, as of 984cbe4f84a6eb32153f7fc46c220a0ccb2746f9

theuni commented:
> c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz

Has been generated and is ready for use now by c-i.

fanquake commented:
> Has been generated and is ready for use now by c-i.

Force pushed for a CI run.

theuni commented on depends/hosts/darwin.mk:
```diff
@@ -71,6 +71,10 @@ $(foreach TOOL,$(cctools_TOOLS),$(eval darwin_$(TOOL) = $$(build_prefix)/bin/$$(
 #
 #         Adds the desired paths from the SDK
 #
+#     -platform_version
```
Does this obsolete/interact with mlinker-version at all?

fanquake replied:
From having another look now, Clang still uses `-mlinker-version`, in it's decision making about whether to use `-platform_version` (by default), which it does if the linker version is > 520. If we are passing `-platform_version` ourselves, then maybe we can drop `-mlinker-version`, but I would have to look again, to see if there are other side effects, and this is probably ok to leave for now, and revisit when we do the lld switch, as I assume more flags are going to change there as well.

Sjors reviewed (commented):
Guix hashes: (will bake new ones for 95163cc later if CI actually passes)

```
fedbab3012043b4277a19fbd0a1abdb5d8e895446bba470981cef361064cdba6  guix-build-984cbe4f84a6/output/arm64-apple-darwin/SHA256SUMS.part
e40c88780560f4aa1e15657373cf6e6149878c90c1acb2980a6015c368378485  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin-unsigned.tar.gz
795221e68c86cb46684dcb80e729e6dd887402930bc8f911b97840f563c4fab8  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin-unsigned.zip
079058ec4d3ddc91092166609f6adc5254a60eae0b2f8787747177c749b97858  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin.tar.gz
cae9d3794b0709f732f7298bc0a969c1a96db9094c64f2de127757b538d935ef  guix-build-984cbe4f84a6/output/dist-archive/bitcoin-984cbe4f84a6.tar.gz
4a6f58cdc176a7d533d4018603f2f20fc6439524ab9ca3d1b4d675e41b2938c2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/SHA256SUMS.part
6483c6749d7568321dfaba94b91d5940f9b6699b3c7816029a6b8eb9e031eaf2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin-unsigned.tar.gz
662d49b36ff53c19f11cdc02319bb4066ee8033343d90f898eeef063be189ed2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin-unsigned.zip
18e670ae7679c0b9317d57412839bd2676b322ae0d18a599bad6c1beddef2413  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin.tar.gz
```

Lightly tested the binary on Intel macOS 13.6.1 and 14.1.2.

theuni reviewed (commented):
Concept ACK. My generated tarball hash matches @fanquake's.

fanquake commented:
Guix build (aarch64):
```bash
fedbab3012043b4277a19fbd0a1abdb5d8e895446bba470981cef361064cdba6  guix-build-984cbe4f84a6/output/arm64-apple-darwin/SHA256SUMS.part
e40c88780560f4aa1e15657373cf6e6149878c90c1acb2980a6015c368378485  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin-unsigned.tar.gz
795221e68c86cb46684dcb80e729e6dd887402930bc8f911b97840f563c4fab8  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin-unsigned.zip
079058ec4d3ddc91092166609f6adc5254a60eae0b2f8787747177c749b97858  guix-build-984cbe4f84a6/output/arm64-apple-darwin/bitcoin-984cbe4f84a6-arm64-apple-darwin.tar.gz
cae9d3794b0709f732f7298bc0a969c1a96db9094c64f2de127757b538d935ef  guix-build-984cbe4f84a6/output/dist-archive/bitcoin-984cbe4f84a6.tar.gz
4a6f58cdc176a7d533d4018603f2f20fc6439524ab9ca3d1b4d675e41b2938c2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/SHA256SUMS.part
6483c6749d7568321dfaba94b91d5940f9b6699b3c7816029a6b8eb9e031eaf2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin-unsigned.tar.gz
662d49b36ff53c19f11cdc02319bb4066ee8033343d90f898eeef063be189ed2  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin-unsigned.zip
18e670ae7679c0b9317d57412839bd2676b322ae0d18a599bad6c1beddef2413  guix-build-984cbe4f84a6/output/x86_64-apple-darwin/bitcoin-984cbe4f84a6-x86_64-apple-darwin.tar.gz
```

fanquake committed:
build: patch boost process for macOS 14 SDK

fanquake commented:
Added the upstream patch to fix Boost Process.

fanquake commented:
Guix Build (aarch64):
```bash
5d111a7c468338e34022cd20103ecd2c130a0d385dcf1d0f0110450b6a71243f  guix-build-b3ab0c3819cf/output/arm64-apple-darwin/SHA256SUMS.part
e9270dbd3358f7d071533142c669ab0f13af63dc1786a9100a43f366539d03c1  guix-build-b3ab0c3819cf/output/arm64-apple-darwin/bitcoin-b3ab0c3819cf-arm64-apple-darwin-unsigned.tar.gz
2af5a689a4f9d7da60003ab0a4844d43239d88c21b984e870f1d8e6cc2d5e28e  guix-build-b3ab0c3819cf/output/arm64-apple-darwin/bitcoin-b3ab0c3819cf-arm64-apple-darwin-unsigned.zip
57cd81bafe5d44439d0e16c6a1767235ea197b3abe8e76997e05e5060ef32dff  guix-build-b3ab0c3819cf/output/arm64-apple-darwin/bitcoin-b3ab0c3819cf-arm64-apple-darwin.tar.gz
3f5a18392f8d0bd7362ed4718384948b24230f8535dcfa641f4adfb038f0a2cb  guix-build-b3ab0c3819cf/output/dist-archive/bitcoin-b3ab0c3819cf.tar.gz
8b1332ddf5126291d2b8ee8d42861268b7f3b80883b211e991deae11a0142bd4  guix-build-b3ab0c3819cf/output/x86_64-apple-darwin/SHA256SUMS.part
34acc961b311626da6d8ed2709b740ee5697ef3e5e8d7ac5cd6a91ac90e9d601  guix-build-b3ab0c3819cf/output/x86_64-apple-darwin/bitcoin-b3ab0c3819cf-x86_64-apple-darwin-unsigned.tar.gz
c196d38ebd34e03d7e5364a4a0f5d1410853ad43c9755fc9ba8921edadf14226  guix-build-b3ab0c3819cf/output/x86_64-apple-darwin/bitcoin-b3ab0c3819cf-x86_64-apple-darwin-unsigned.zip
2cc8083a57947774a7c1bd0d13124b6464a91b6ec9239c96d9cddd543ed06675  guix-build-b3ab0c3819cf/output/x86_64-apple-darwin/bitcoin-b3ab0c3819cf-x86_64-apple-darwin.tar.gz
```

Sjors commented:
Why did the Boost issue stop CI, but not the Guix build?

fanquake commented:
Because the CI explicitly passes --enable-external-signer.

Okponko commented:
![Uploading IMG_3704.jpeg…]()

Okponko commented:
The Shop right here

hebasto reviewed (approved):
ACK 8ea45e626e5a0482ee11d4376f961d8126ce7c7b.

I've got the same hashes for SDK on both platforms:
- macOS:
```
% shasum -a 256 Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz 
c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```
- Linux:
```
$ sha256sum Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz 
c0c2e7bb92c1fee0c4e9f3a485e4530786732d6c6dd9e9f418c282aa6892f55d  Xcode-15.0-15A240d-extracted-SDK-with-libcxx-headers.tar.gz
```

My Guix build hashes are the same for `x86_64` and `aarch64`:
```
cb84be46cc7139072879655a87a36303027ee1faefbf251e5827ee69220458cf  guix-build-8ea45e626e5a/output/arm64-apple-darwin/SHA256SUMS.part
e064e0380ffa84a0da5e6196d59639011f7860bf61b13c2626e9fdacd0e6ebc0  guix-build-8ea45e626e5a/output/arm64-apple-darwin/bitcoin-8ea45e626e5a-arm64-apple-darwin-unsigned.tar.gz
7ca6da72095dc2fcf9ad30fcebffb4d5de58bc26c9f890eec34067610d55472f  guix-build-8ea45e626e5a/output/arm64-apple-darwin/bitcoin-8ea45e626e5a-arm64-apple-darwin-unsigned.zip
dede67b0ebbb9d05703748aa8e2819f30503d20da22b56503922005a79ddc81e  guix-build-8ea45e626e5a/output/arm64-apple-darwin/bitcoin-8ea45e626e5a-arm64-apple-darwin.tar.gz
8f1dcce3498f37485e6cccaeacc61d505d31542ecfd2f345eb785562e95ee434  guix-build-8ea45e626e5a/output/dist-archive/bitcoin-8ea45e626e5a.tar.gz
852cdf940cd26f5d1be65f95405585950d32052be0be8ea859e6b7596d277d9a  guix-build-8ea45e626e5a/output/x86_64-apple-darwin/SHA256SUMS.part
6b2a666ceb784d4bc740df0b425c2aa42af8d1d44b44cd338ab360ee42ccc5ab  guix-build-8ea45e626e5a/output/x86_64-apple-darwin/bitcoin-8ea45e626e5a-x86_64-apple-darwin-unsigned.tar.gz
c982ef2ce2c55113b46cfbe8312f4ad01eea24935bbaa19bc26ca8aaed329345  guix-build-8ea45e626e5a/output/x86_64-apple-darwin/bitcoin-8ea45e626e5a-x86_64-apple-darwin-unsigned.zip
06433a4f20a037f738fd5de5d8928442e32e3671ffe4878bed18aa322bade0f9  guix-build-8ea45e626e5a/output/x86_64-apple-darwin/bitcoin-8ea45e626e5a-x86_64-apple-darwin.tar.gz
```

Guix builds are configured properly:
```
Options used to compile and link:
  external signer = yes
```

---

Can we get rid of linker warnings:
```
ld: warning: passed two min versions (11.0, 11.0) for platform macOS. Using 11.0.
```
?

fanquake commented:
> Can we get rid of linker warnings:

I assume this is a combination of `-mmacosx-version-min`, `-Wl,-platform_version` and maybe cctools. Note that the warnings have also already been appearing in our security checks/tests, for some time. This should dissapear when we switch to lld, so it might not be worth making more changes now.

sedited reviewed (approved):
ACK 8ea45e626e5a0482ee11d4376f961d8126ce7c7b

Get the same build hashes and SDK hashes. Ran the binary on macOS 11.1. Completed a depends build on my linux dev box. Verified the boost process patch.
